src/custom_types/scroll_menu.py

```diff
@@ -181,10 +184,26 @@ def _display_content(
                     ]
                 )
             )
+
+            # Change the header (selected path)
+            self.body.children.pop(0)
+            self.modify_header(self.path)
```

Comment:
FYI, it would be a better pattern here to have `self.modify_header` return a value and then you mutate `self.text` inline here. As written this code is hard to follow.

I don't think it is necessary to make this change if you don't want to though.